runtime/lua/man.lua

```diff
@@ -1,7 +1,66 @@
 require('vim.compat')
 
+local api, fn = vim.api, vim.fn
+
+local find_arg = '-w'
+local localfile_arg = true -- Always use -l if possible. #6683
 local buf_hls = {}
 
+local M = {}
+
+local function man_error(msg)
+  M.errormsg = 'man.lua: ' .. vim.inspect(msg)
+  error(M.errormsg)
+end
+
+-- Run a system command and timeout after 30 seconds.
+local function man_system(cmd)
+  local opts = {
+    stdout = '',
+    stderr = '',
+    exit_code = 0,
+  }
+
+  local function system_handler(_, data, event)
+    if event == 'stdout' or event == 'stderr' then
+      opts[event] = opts[event] .. table.concat(data, '\n')
+    else
+      opts.exit_code = data
+    end
+  end
+
+  opts.on_stdout = system_handler
+  opts.on_stderr = system_handler
+  opts.on_exit = system_handler
+  local jobid = fn.jobstart(cmd, opts)
+
+  if jobid < 1 then
+    man_error(string.format('command error %d: %s', jobid, table.concat(cmd)))
+  end
+
+  local res = fn.jobwait({ jobid }, 30000)
+  if res[1] == -1 then
+    pcall(function()
+      fn.jobstop(jobid)
+      man_error(string.format('command timed out: %s', table.concat(cmd, ' ')))
+    end)
```

Comment:
In original code only `E900:` error is suppressed. Other errors are still thrown.